cdap-common/src/main/java/io/cdap/cdap/common/internal/remote/RemoteClient.java

```diff
@@ -155,14 +166,24 @@ public HttpURLConnection openConnection(String resource) throws IOException {
     urlConn.setDoInput(true);
     RemoteAuthenticator authenticator = getAuthenticator();
     if (authenticator != null) {
-      urlConn.setRequestProperty(HttpHeaders.AUTHORIZATION,
-                                 String.format("%s %s", authenticator.getType(), authenticator.getCredentials()));
+      urlConn.setRequestProperty(HttpHeaders.AUTHORIZATION, String.format("%s %s", authenticator.getType(),
```

Comment:
That's a quite large piece of dup code. Please extract and reuse. You can pass `BiConsumer<String, String>` in to fill headers/requestProperties as needed.

Reply:
I'm a bit unclear on how you want to use BiConsumer here since it does not return a value, but I went ahead and created a helper function to reduce duplication.